api/filter_outputs.go

```diff
@@ -325,10 +315,37 @@ func (api *FilterAPI) addEventHandler(w http.ResponseWriter, r *http.Request) {
 	log.Info(ctx, "added event to filter output", logData)
 }
 
-func (api *FilterAPI) addEvent(ctx context.Context, filterOutputID string, event *models.Event) error {
-	if event.Type == "" {
-		return filters.NewBadRequestErr("event type cannot be empty")
+func (api *FilterAPI) addEvents(ctx context.Context, events []*models.Event, id string, completed bool) error {
+	logData := log.Data{"filter_output_id": id}
+
+	if len(events) > 0 {
+		for _, e := range events {
+			if err := e.Validate(); err != nil {
+				log.Error(ctx, "filter output event failed validation", err, logData)
+				return filters.NewBadRequestErr(err.Error())
+			}
+
+			if err := api.dataStore.AddEventToFilterOutput(ctx, id, e); err != nil {
+				log.Error(ctx, "failed to add event to filter output", err, logData)
+				return err
```

Comment:
Should we be logging errors and then returning them? We end up with the same error logged multiple times?

Reply:
Good point, have removed